packages/expo-doctor/src/checks/GlobalPrereqsVersionCheck.ts

```diff
@@ -0,0 +1,113 @@
+import spawnAsync from '@expo/spawn-async';
+import { execSync } from 'child_process';
+import semver from 'semver';
+
+import { DoctorCheck, DoctorCheckParams, DoctorCheckResult } from './checks.types';
+
+const MIN_WATCHMAN_VERSION = '4.6.0';
+const MIN_NPM_VERSION = '3.0.0';
+const CORRECT_NPM_VERSION = 'latest';
+const WARN_NPM_VERSION_RANGES = ['>= 5.0.0 < 5.7.0'];
+const BAD_NPM_VERSION_RANGES = ['>= 5.0.0 <= 5.0.3'];
+
+function _isNpmVersionWithinRanges(npmVersion: string, ranges: string[]) {
+  return ranges.some(range => semver.satisfies(npmVersion, range));
+}
+
+async function _checkNpmVersionAsync(): Promise<string | null> {
+  try {
+    try {
+      const yarnVersionResponse = await spawnAsync('yarnpkg', ['--version']);
+      if (yarnVersionResponse.status === 0) {
+        return null;
+      }
+    } catch {}
+
+    const npmVersion = execSync('npm --version', { stdio: 'pipe' }).toString().trim();
+
+    if (
+      semver.lt(npmVersion, MIN_NPM_VERSION) ||
+      _isNpmVersionWithinRanges(npmVersion, BAD_NPM_VERSION_RANGES)
+    ) {
+      return `Error: You are using npm version ${npmVersion}. We recommend the latest version ${CORRECT_NPM_VERSION}. To install it, run 'npm i -g npm@${CORRECT_NPM_VERSION}'.`;
+    } else if (_isNpmVersionWithinRanges(npmVersion, WARN_NPM_VERSION_RANGES)) {
+      return `Warning: You are using npm version ${npmVersion}. There may be bugs in this version, use it at your own risk. We recommend version ${CORRECT_NPM_VERSION}.`;
+    }
+  } catch {
+    return `Warning: Could not determine npm version. Make sure your version is >= ${MIN_NPM_VERSION} - we recommend ${CORRECT_NPM_VERSION}.`;
+  }
+
+  return null;
+}
+
+async function _checkWatchmanVersionAsync(): Promise<string | null> {
+  const homebrewInstallNote = `\n\nIf you are using homebrew, try:\nbrew uninstall watchman; brew install watchman`;
+
+  let watchmanVersion;
+  // There's no point in checking any of this stuff if watchman isn't supported on this platform
+  if (process.platform !== 'darwin') {
+    return null;
+  }
+
+  // all checks going forward assume macOS, which should have watchman
+
+  const cannotDetectWatchmanInstallationMessage =
+    'Cannot detect watchman installation.' + homebrewInstallNote;
+
+  try {
+    watchmanVersion = JSON.parse((await spawnAsync('watchman', ['version'])).stdout.trim()).version;
+  } catch (e: any) {
+    return cannotDetectWatchmanInstallationMessage;
+  }
+
+  // If we can't get the watchman version, `getVersionAsync` will return `null`
```

Comment:
What is `getVersionAsync`, and if it doesn't exist anymore, do we need watchman at all?